pkg/controller/node/node_controller.go

```diff
@@ -979,6 +983,27 @@ func (ctrl *Controller) updateCandidateMachines(pool *mcfgv1.MachineConfigPool,
 	return nil
 }
 
+// sortNodeList sorts the list of candidate nodes by label topology.kubernetes.io/zone
+// nodes without label are at end of list and sorted by age (oldest to youngest)
+func sortNodeList(nodes []*corev1.Node) []*corev1.Node {
+	sort.Slice(nodes, func(i, j int) bool {
+		_, iOk := nodes[i].Labels["topology.kubernetes.io/zone"]
+		_, jOk := nodes[j].Labels["topology.kubernetes.io/zone"]
+		// if both nodes have zone label, sort by zone, push nodes without label to end of list
+		if iOk && jOk {
+			return nodes[i].Labels["topology.kubernetes.io/zone"] < nodes[j].Labels["topology.kubernetes.io/zone"]
```

Comment:
Why not also retrieve the value above, like:
`iZone, iOk := nodes[i].Labels["topology.kubernetes.io/zone"]`
then here do `return iZone < jZone` or so?

Reply:
I'm net neutral on this so will do